### ISSUE TITLE
`hits_dec` inference samples every token from the last position

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#14]
`inference()` pre-fills the whole target sequence, then repeatedly reads the logits at the **last** position while writing to position `ix`:

```python
# scripts/modeling/hits_dec/train.py:70-100
y = torch.tensor([[pad_ix] * n_tokens], ...)   # full length up front

for ix in range(n_tokens):
    logits = model(y)
    logits = logits[:, -1, :]      # always position n_tokens-1
    ...
    y[:, ix] = y_next.item()       # written to position ix
```

So every token is sampled from the distribution at the final position, conditioned on a sequence that is still mostly pad/garbage — instead of the usual "grow the sequence, read the last position" loop. The model is never asked to predict position `ix` from `y[:ix]`.

Same loop in `scripts/modeling/hits_dec_xformer/train.py:64-81`.

`rhythmic_relationships/evaluate.py:hits_inference` and `hits_encdec_polydrums/train.py:inference` do it correctly (start with `[[start_ix]]` and `torch.cat` each new token), so the fix is to follow those.

**Fix**: either grow `y` from a single start token and keep taking `[:, -1, :]`, or keep the pre-filled buffer and read `logits[:, ix, :]`. The first matches the rest of the codebase.
